### DISCUSSION TITLE
Is it possible to export a custom MLX model to run anywhere else than Apple Silicon?

### DISCUSSION THREAD

**Discussion by FerusAndBeyond** [ml-explore/mlx#1730] [Q&A]
I'm interested in using multiple physical devices for training ML models using MLX, then export it to a format where I can run it anywhere, e.g. on a server/linux. This is not a fixed model from huggingface but a custom model trained from scratch on custom data. 

Is it possible? I couldn't find anything at all, just discussions about ONNX.

**Comment by awni** [comment]
No not really. You can build and run MLX on a linux server and it should work (no GPU support just CPU). There is no export to other framework functionality like ONNX or PyTorch etc. Is exporting to ONNX what you are looking for or is it something else?

**Reply by FerusAndBeyond** [reply]
Okay, didn't know about the CPU support. That should actually be fine in my case as inference speed isn't critical for the application in mind. But ONNX as you mentioned would be ideal. Of course, if I can export to, say PyTorch, I should be able to create ONNX from it.
